constantine/commitments/ipa/multiproof.nim

```diff
@@ -0,0 +1,337 @@
+# Constantine
+# Copyright (c) 2018-2019    Status Research & Development GmbH
+# Copyright (c) 2020-Present Mamy André-Ratsimbazafy
+# Licensed and distributed under either of
+#   * MIT license (license terms in the root directory or at http://opensource.org/licenses/MIT).
+#   * Apache v2 license (license terms in the root directory or at http://www.apache.org/licenses/LICENSE-2.0).
+# at your option. This file may not be copied, modified, or distributed except according to those terms.
+
+import
+  tables,
+  ./[transcript_gen, common_utils, ipa_prover, barycentric_form, helper_types],
+  ../../../constantine/platforms/primitives,
+  ../../../constantine/hashes,
+  ../../math/config/[type_ff, curves],
+  ../../../constantine/math/elliptic/[ec_multi_scalar_mul, ec_multi_scalar_mul_scheduler],
+  ../../math/elliptic/[ec_twistededwards_projective, ec_twistededwards_batch_ops],
+  ../../../constantine/math/arithmetic,
+  ../../../constantine/platforms/[views],
+  ../../../constantine/math/io/[io_bigints,io_fields],
+  ../../../constantine/curves_primitives
+
+
+# ############################################################
+#
+#                   Multiproof System
+#
+# ############################################################
+
+# The multiproof is a multi-proving system for several polynomials in the evaluation form
+
+# Converts the const DOMAIN 256 to Fr[Banderwagon]
+func domainToFrElem* [EC_P_Fr] (res: var EC_P_Fr, inp: uint8)=
```

Comment:
This conflicts with helper_types defined `EC_P_Fr`

https://github.com/mratsim/constantine/blob/15f9523a2592900ea4932b9be9835601e9579a77/constantine/commitments/ipa/helper_types.nim#L23-L26

Either use

```suggestion
func domainToFrElem*[Field](res: var Field, inp: uint8)=
```

Or because it's constrained to Fr, use

```suggestion
func domainToFrElem*(res: var Fr, inp: uint8)=
```

Reply:
Switched to `Fr`